### PR TITLE
download from list of links in a text message

### PR DESCRIPTION
**Pull request by SpEcHiDe**

### CODE REVIEW

**Comment by Krishna-Singhal** [comment]
No need for check for media [here](https://github.com/UsergeTeam/Userge/blob/6b8771e2d21a5d97a01b76ad609be99df966801d/userge/plugins/misc/download.py#L36)

File: userge/plugins/misc/download.py [hunk 1]
```diff
@@ -33,15 +33,15 @@
...
-    if message.reply_to_message and message.reply_to_message.media:
+    if message.reply_to_message:
         resource = message.reply_to_message
     elif message.input_str:
         resource = message.input_str
     else:
         await message.err("nothing found to download")
         return
     try:
-        dl_loc, d_in = await handle_download(message, resource)
```
Comment (Krishna-Singhal) [inline userge/plugins/misc/download.py:44]: why did you remove these vars

File: userge/plugins/misc/download.py [hunk 2]
```diff
@@ -149,6 +149,20 @@ async def tg_download(
     message: Message, to_download: Message, from_url: bool = False
 ) -> Tuple[str, int]:
     """ download from tg file """
+    if not message.media:
+        ets = message.entities or message.caption_entities or []
+        text = message.text or message.caption or ""
```
Comment (Krishna-Singhal) [inline userge/plugins/misc/download.py:154]: here should be

```python
if not to_download.media:
    ets = to_download.entities or to_download.caption_entities or []
    text = to_download.text or to_download.caption or ""
```

**Review by Krishna-Singhal — CHANGES_REQUESTED** [review]

File: userge/plugins/misc/download.py (hunk 2 above)
Comment (SpEcHiDe) [reply userge/plugins/misc/download.py:154]: why? 🤔😅

File: userge/plugins/misc/download.py (hunk 1 above)
Comment (SpEcHiDe) [reply userge/plugins/misc/download.py:44]: It was issue with my  previous commit, 

try this 

.download http://speed.hetzner.de/100MB.bin

then check logs after downloading is completed 🤒🤕🤕

File: userge/plugins/misc/download.py (hunk 2 above)
Comment (Krishna-Singhal) [reply userge/plugins/misc/download.py:154]: > why? 🤔😅

resource is to_download not message 🤔

File: userge/plugins/misc/download.py (hunk 1 above)
Comment (Krishna-Singhal) [reply userge/plugins/misc/download.py:44]: > It was issue with my previous commit,
> 
> try this
> 
> .download http://speed.hetzner.de/100MB.bin
> 
> then check logs after downloading is completed 🤒🤕🤕

cuz you are not returning anything if downloading list of urls

File: userge/plugins/misc/download.py [hunk 3]
```diff
@@ -151,19 +151,10 @@ async def tg_download(
...
-                length = entity.length
-                url = text[offset:offset + length]
-            if url:
-                dl_loc, b_ = await url_download(message, url)
-                mite += b_
+        ets = extract_urls(to_download)
+        for uarl in ets:
+            dl_loc, b_ = await url_download(message, uarl)
+            mite += b_
         return dl_loc, mite
```
Comment (Krishna-Singhal) [inline userge/plugins/misc/download.py:158]:
```python
    if not to_download.media:
        dl_loc, mite = [], 0
        ets = extract_urls(to_download)
        if len(ets) == 0:
            raise Exception("nothing found to download")
        for uarl in ets:
            _dl_loc, b_ = await url_download(message, uarl)
            dl_loc.append(_dl_loc)
            mite += b_
        return dumps(dl_loc), mite
```

**Review by Krishna-Singhal — CHANGES_REQUESTED** [review]

**Review by Krishna-Singhal — APPROVED** [review]

**Comment by rking32** [comment]
Hi @SpEcHiDe and @Krishna-Singhal, Thanks for this and merging ... :)
